match upstream default setting for editor.editContext (enabled)

sharon-wang opened the pull request:
### Summary

- addresses https://github.com/posit-dev/positron/issues/9327
- approach: update default `editor.editContext` setting to match upstream, to fix output copying issue. If this setting is disabled in VS Code, the same issue occurs there as well.
- small code change, but has widespread implications for e2e tests as the target locators and corresponding methods now need to be compatible with the `.native-edit-context` div, instead of a `textarea`

    <details><summary>Range of tests failing (hoping to fix these in this PR)</summary>
    
    Essentially anything that is backed with a monaco editor involving copy or paste, so Console, Notebook, Connections, Output, Assistant chat, etc.
    
    <img width="303" height="264" alt="image" src="https://github.com/user-attachments/assets/a20e97f4-a3a9-4882-a7ba-2a256a9dda65" />
    
    <img width="306" height="423" alt="image" src="https://github.com/user-attachments/assets/bbbf0a43-096b-4150-8061-b35dce2d3b61" />
    
    <img width="343" height="423" alt="image" src="https://github.com/user-attachments/assets/a126b8e6-0abc-40ae-b9c8-d8046300d3de" />
    
    </details> 

### Release Notes

#### Bug Fixes

- Fix issue preventing text from being copied from Output pane (#9327)

### QA Notes

@:output

- added a test to copy ~1000 characters from the output pane and paste to an editor
- the test is skipped because the assertion is failing in CI, though passing locally for me. Seems like an issue with newlines/normalization getting in the way of the string match

This setting was originally disabled by default for this reason:
> // Disable the edit context which is normally enabled.
> // It causes test failures due to changes in the DOM.

This was when the setting was still experimental, and the setting was called `editor.experimentalEditContextEnabled` during that time

This setting was migrated to be non-experimental (with the current setting name `editor.editContext`) in 1.102: https://github.com/microsoft/vscode/commit/312284e7db0d90894a576a67c0aaf07319fc470b

Now that we are enabling `editor.editContext` by default, locators using `textarea` have been updated to `.native-edit-context`, with the exception of `test/e2e/pages/search.ts`, which seems to still actually use a textarea.

Because of this change from `textarea` --> `.native-edit-context` div, we can no longer call `.fill()` on the locators, because `Error: locator.fill: Error: Element is not an <input>, <textarea> or [contenteditable] element`, so `.fill()` has been replaced with `.pressSequentially()`.

sharon-wang committed:
match upstream default setting for editor.editContext (enabled)

update default editor.editContext setting to fix output copying issue

if this setting is disabled in VS Code, the same issue occurs there as well

juliasilge reviewed (dismissed):
This fixes the problem!!! 🥹🚀🎉  

What would you think about running quite a robust suite of tests against this, given how we ended up in this situation? Maybe the "Full Suite" GH action? And we may need to loop in an expert on the e2e tests to resolve problems we see.

I'm approving but let's make sure that we are quite confident about the impact to tests before merging.

sharon-wang committed:
update console pasteInMonaco to use editContext locator

sharon-wang committed:
update editors selector in waitForActiveEditor to use native edit context

sharon-wang committed:
add test for copying from output to editor

sharon-wang committed:
replace more textarea -> .native-edit-context and .fill() -> .pressSequentially()

fill can't be used for the native edit context, so let's swap to pressSequentially. See error when using .fill() on a native edit context locator: `Error: locator.fill: Error: Element is not an <input>, <textarea> or [contenteditable] element`

sharon-wang committed:
move normalize function to util

sharon-wang committed:
normalize copied text before assertion for output-to-editor-copy.test.ts

this is passing locally, but failing in CI and looks to be due to the newlines not being normalized

sharon-wang committed:
Revert "normalize copied text before assertion for output-to-editor-copy.test.ts"

This reverts commit 36d709148e810de2c7f4c68beccae09d6bec959c.

sharon-wang committed:
Revert "move normalize function to util"

This reverts commit b3015e3dc36d53bbdf2cd662d112912d308dacd6.

sharon-wang committed:
Skip 'Output to Editor Copy' test due to CI assertion failure

testlabauto reviewed (approved):
Looks good from the test standpoint. Thank you for taking care of the test related items!
